WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormRemoteActionUseCase.swift

```diff
@@ -102,6 +102,24 @@ final class ProductFormRemoteActionUseCase {
             }
         }
     }
+
+    /// Delete a product remotely.
+    /// - Parameters:
+    ///   - product: The product to be deleted remotely.
+    ///   - onCompletion: Called when the remote process finishes.
+    func deleteProduct(product: EditableProductModel, onCompletion: @escaping EditProductCompletion) {
+        deleteProductRemotely(product: product) { productResult in
+            switch productResult {
+            case .failure(let error):
+                // TODO: M5 analytics
+                onCompletion(.failure(error))
+            case .success(let product):
+                // `self` is retained because the use case is not usually strongly held.
```

Comment:
nit: it looks like `self` isn't used in this block 😄 so maybe this comment isn't needed anymore?